src/components/FeatureViews/Scheduler/Scheduler.js

```diff
@@ -200,38 +243,83 @@ class Scheduler extends Component {
     changeView = (value) => {
         let calendarApi = this.calendarComponentRef.current.getApi();
         calendarApi.changeView(value);
+        let filter;
+
         let date = this.currentDate();
-        this.setState({
-            "viewValue": value,
-            "currentDate": date,
+        this.setState(() => {
+            this.props.calendarActions.fetchSessions({
+                config: {
+                    params: {
+                        time_frame: filter,
+                        view_option: this.viewOptions[this.state.calendarFilterValue],
+                    }
+                }
+            });
```

Comment:
Move this outside the `setState`, as it is not involved in setting the state. Also, filter does not appear to be initialized.